molecule/config.py

```diff
@@ -0,0 +1,106 @@
+        :return: None
+        """
+        # load defaults from provided file
+        if defaults_file is None:
+            defaults_file = os.path.join(os.path.dirname(__file__), 'conf/defaults.yml')
+
+        with open(defaults_file, 'r') as stream:
+            self.config = yaml.load(stream)
+
+    def merge_molecule_config_files(self, paths=CONFIG_PATHS):
+        """
+        Looks for a molecule config file in paths and merges it with current config if found
+
+        Only the first file that's found will be merged in.
+        :param paths: list of places to look for config files
+        :return: Path of file that was merged into config, if found, otherwise None
+        """
+        # merge defaults with a config file if found
+        for path in paths:
+            if path and os.path.isfile(path):
+                with open(path, 'r') as stream:
+                    self.config = utilities.merge_dicts(self.config, yaml.load(stream))
+                    return path
+        return
+
+    def merge_molecule_file(self, molecule_file=None):
+        """
+        Looks for a molecule file in the local path and merges it into our config
+
+        :param molecule_file: path and name of molecule file to look for
+        :return: None
+        """
+        if molecule_file is None:
+            molecule_file = self.config['molecule']['molecule_file']
+
+        if not os.path.isfile(molecule_file):
+            error = '\n{}Unable to find {}. Exiting.{}'
+            print(error.format(Fore.RED, self.config['molecule']['molecule_file'], Fore.RESET))
+            sys.exit(1)
+
+        with open(molecule_file, 'r') as env:
+            try:
+                molecule_yml = yaml.load(env)
+            except Exception as e:
+                error = "\n{}{} isn't properly formatted: {}{}"
+                print(error.format(Fore.RED, molecule_file, e, Fore.RESET))
+                sys.exit(1)
+
+            interim = utilities.merge_dicts(self.config, molecule_yml)
+            self.config = interim
+
+    def build_easy_paths(self):
+        """
+        Convenience function to build up paths from our config values
+
+        :return: None
+        """
+        self.config['molecule']['state_file'] = '/'.join([self.config['molecule']['molecule_dir'], self.config[
+            'molecule']['state_file']])
+        self.config['molecule']['vagrantfile_file'] = '/'.join([self.config['molecule']['molecule_dir'], self.config[
+            'molecule']['vagrantfile_file']])
+        self.config['molecule']['rakefile_file'] = '/'.join([self.config['molecule']['molecule_dir'], self.config[
+            'molecule']['rakefile_file']])
+        self.config['ansible']['config_file'] = '/'.join([self.config['molecule']['molecule_dir'], self.config[
+            'ansible']['config_file']])
+        self.config['ansible']['inventory_file'] = '/'.join([self.config['molecule']['molecule_dir'], self.config[
+            'ansible']['inventory_file']])
```

Comment:
If you are building paths, you must use ``os.path.join`` - https://docs.python.org/2/library/os.path.html.